WordPress/gutenberg
# Video Block: add raw transformation from `video` html

Closes: #42768

## What?
This PR adds a transformation that generates a video block from HTML containing a `video` element.

## Why?
Since `core/image` and `core/embed` support transformation from raw HTML, it would make sense to add similar support for the video block.

## How?

Inherits as many attributes supported by the Video block as possible, including the `src` attribute of the `video` element. However, the `src` attribute is assumed to be attached to the `video` tag, and the `source` tag inside the `video` tag is not considered.

This is because the Clipboard API itself doesn't seem to pick up elements inside the `video` tag. The screencast below shows that when you copy HTML containing a `video` element with a `source` element, the clipboard does not contain the `source` element.

https://github.com/WordPress/gutenberg/assets/54422211/dbf5efc3-c231-44cd-822a-f54069747104

## Testing Instructions

- Open the Codepen sample page prepared [here](https://codepen.io/Tetsuaki-Hamno/pen/NWJqZKR).
- Copy the entire page and paste it onto the block editor.
- It should inherit the attributes and be converted to a video block.

## Screencast

https://github.com/WordPress/gutenberg/assets/54422211/4782fa67-88d5-4e90-88b9-3181345a7fa7
